ledger/sandbox/src/main/resources/db/migration/postgres/V34__Parties_is_local.sql

```diff
@@ -0,0 +1,21 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+---------------------------------------------------------------------------------------------------
+-- V34: Add is_local column to parties table.
+--
+-- This allows parties to be marked as to whether they were added locally as determined by whether
+-- the specified participant_id matched the local participant_id.
+--
+-- To initialize parties.is_local properly, rely on the existence of party_entries.is_local
+-- default to parties.is_local = true only if party_entries were somehow manually deleted.
```

Comment:
I'm not sure I understand how the party can be missing. Can you clarify what you mean by "somehow"?

Reply:
I meant if someone manually connected to the index database and removed rows from the `party_entries` table, e.g. to free up space. I just felt wary about setting is_local to true in the case no row is found in `party_entries` and therefore mislabeling.

Reply:
The fact that the index is backed by an RDBMS is an internal detail. If you start manually mangling with it instead of interacting with it with a public API you are asking for trouble. I think we should crash if the index has been corrupted.

Reply:
Ok, letting the final `ALTER TABLE .. SET NOT NULL` command fail if there are uninitialized values.